extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -953,11 +961,69 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 		return getLocationDisplayName(location);
 	}
 
-	public async getAzureResourceGroupDropdownValues(subscription: azureResource.AzureResourceSubscription): Promise<azdata.CategoryValue[]> {
+	public async getAzureResourceGroupDropdownValues(resourceType: AzureResourceType, subscription: azureResource.AzureResourceSubscription): Promise<azdata.CategoryValue[]> {
```

Comment:
I suggest making 4 different methods and not use switch

Reply:
Done, separated into 4 different methods. Added back the original one though (`getAzureResourceGroupDropdownValues`, without a specific resource type) since we still need the generic "list all resource groups" functionality for the "create new DMS" dialog